### DISCUSSION TITLE
Multiple Configuration Folders

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#1525] [Q&A] [answered]
Hi,

One of the components I am writing has a configuration header filer, similar to multiple other fprime components. However, since I have the fprime folder as a submodule in my repository, I cannot place the header file in the same config file as the rest of the fprime components. 

Is it possible to point fprime to look at multiple configuration folders? I tried to set the `config_directory` item in my deployment `settings.ini` file to be a colon-separated list of paths(similar to `library_locations`) however, it does not seem to be searching both locations, only the first one listed.

**Comment by kevin-f-ortega** [comment, marked as answer]
You can do the following:

1. Add a config directory in your project directory, let's call this directory `config`
2. Add your config file to `config`, let's call this config file MyConfigFile.hpp
3. In your project's CMakeLists.txt, add this line `include_directories("${CMAKE_CURRENT_LIST_DIR}/config")`. You may need to update this path to use the correct path to `config`
4. In your component, include the config file using `#include "MyConfigFile.hpp"`.

You will now be able to use fprime's config directory plus your config directory.

**Reply by codeflight1** [reply]
While this will work for hpp headers, I think there should be a system in place for fpp config files.

**Reply by ebenezerarunkumar** [reply]
Thanks for that help @kevin-f-ortega! Following up on @codeflight1's question, is there a way to define a fpp config file in the project directory and reference that fpp file in the component directory?

**Reply by kevin-f-ortega** [reply]
I'm not familiar how to do that with fpp config files. @LeStarch, do you know?

**Reply by ebenezerarunkumar** [reply]
Also, 
If we wanted to change the configuration of a fprime component(ex. increase the `STATIC_MEMORY_ALLOCATION_SIZE` for the Static Memory Component), how do we do that when we have the fprime repo as a submodule? Do we have to copy that folder to our repo and point our deployment settings.ini file to look there? 

Having the settings.ini look at multiple config directories could also be helpful in this case.

**Reply by Joshua-Anderson** [reply]
@EbenezerA99 The current recommendation is to copy the fprime config directory into your own deployment and use the settings.ini file to point to it. I agree that being able to supply your own config directory that'll mask any overlapping config in the fprime framework is probably the ideal solution long term.

**Reply by ebenezerarunkumar** [reply]
@Joshua-Anderson Thanks for that info!
